backend/core/models/models.py

```diff
@@ -124,25 +140,32 @@ class model_getconfig(BaseModel):
     webhook: Optional[model_webhook] = None
     queue_strategy: Optional[queue_strat] = None
     post_checks: Optional[List[model_generic_pre_post_check]] = None
+    cache: Optional[model_cache_config] = {}
```

Comment:
- model_cache_config can we add this to the newer getconfig models aswell. 
  This allows more the newer more verbose API routes to support the caching capability.
	- model_napalm_getconfig
	- model_ncclient_getconfig
	- model_netmiko_getconfig
	- model_restconf

Reply:
That's just down to "this is a POC" and wanted to make sure the general approach was okay before taking the time to integrate in the other ones.

I didn't try for very long, but I actually couldn't get model_netmiko_getconfig working at all (with or without caching).  Are you sure those tests pass without this PR?  If so I'll dig in.  Might need to bite the bullet and take the time to setup lab.